torch/csrc/Generator.cpp

```diff
@@ -72,9 +76,12 @@ static PyObject* THPGenerator_pynew(
     self->cdata = make_generator<MPSGeneratorImpl>();
   }
 #endif
+#ifdef USE_XPU
   else if (device.type() == at::kXPU) {
-    self->cdata = at::detail::getXPUHooks().getXPUGenerator(device.index());
-  } else if (device.type() == at::kIPU) {
+    self->cdata = make_generator<XPUGeneratorImpl>(device.index());
```

Comment:
The old version via hooks and less ifdef sounds better?

Reply:
Using hooks instead of ifdef.